miden-lib/asm/sat/internal/tx.masm

```diff
@@ -0,0 +1,110 @@
+use.miden::sat::internal::account
+use.miden::sat::internal::asset
+use.miden::sat::internal::constants
+use.miden::sat::internal::epilogue
+use.miden::sat::internal::layout
+
+#! Returns the block hash of the last known block at the time of transaction execution.
+#!
+#! Inputs: []
+#! Outputs: [H]
+#!
+#! H is the last known block hash.
+export.get_block_hash
+    exec.layout::get_blk_hash
+end
+
+#! Returns the block number of the last known block at the time of transaction execution.
+#!
+#! Inputs: []
+#! Outputs: [num]
+#!
+#! num is the last known block number.
+export.get_block_number
+    exec.layout::get_blk_num
+end
+
+#! Returns the input notes hash. This is computed as a sequential hash of (nullifier, script_root)
+#! tuples over all input notes.
+#!
+#! Inputs: []
+#! Outputs: [COM]
+#!
+#! COM is the input notes hash.
+export.get_input_notes_hash
+    exec.layout::get_nullifier_com
+end
+
+#! Returns the output notes hash. This is computed as a sequential hash of (note_hash, note_metadata)
+#! tuples over all output notes.
+#!
+#! Inputs: []
+#! Outputs: [COM]
+#!
+#! COM is the output notes hash.
+export.get_output_notes_hash
+    exec.epilogue::process_created_notes
+end
```

Comment:
This could probably be a re-export.

Also, is `exec.epilogue::process_created_notes` the right procedure to call here?

Reply:
Yes I believe this is the correct procedure. Created notes can be changed at any point during tx execution and as such we process the re-process the commitment every time this procedure is invoked.  Maybe we should rename the procedure to reflect this?  `compute_output_notes_hash`?

Reply:
Does `process_created_notes` just compute output note hash? Or does it do other things? I guess looking at this it is not immediately clear that it is OK to invoke `process_created_notes` multiple times.

If the only thing it does is compute the hash - then I think we should rename it. But if it does other things, we could probably just add a comment here explaining why it is OK to invoke this procedure multiple times.

Reply:
Yes the only thing it does is compute the hash of the output notes.